File: android/app/src/main/java/com/thankful/app/data/OrderPolling.kt
package com.thankful.app.data

import kotlinx.coroutines.delay

class OrderPollingService(private val orders: OrdersRepository) {
  suspend fun waitUntilPaid(orderId: String, timeoutSec: Int = 30): Boolean {
    val start = System.currentTimeMillis()
    while (System.currentTimeMillis() - start < timeoutSec * 1000) {
      try {
        val (body, code) = orders.getOrder(orderId)
        if (code == 200 && body.contains("\"status\":\"paid\"")) return true
      } catch (_: Exception) {}
      delay(1000)
    }
    return false
  }
}

File: ios/ThankfulApp/Core/OrderPolling.swift
import Foundation

final class OrderPollingService {
  private let orders: OrdersRepository
  init(orders: OrdersRepository) { self.orders = orders }
  func waitUntilPaid(orderId: String, timeoutSec: Int = 30) async -> Bool {
    let start = Date()
    while Date().timeIntervalSince(start) < Double(timeoutSec) {
      do {
        let o = try await orders.getOrder(id: orderId)
        if (o["status"] as? String) == "paid" { return true }
      } catch {}
      try? await Task.sleep(nanoseconds: 1_000_000_000)
    }
    return false
  }
}


